app/interactors/schedule/save_datetime_interactor.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+class Schedule::SaveDatetimeInteractor
+  include Interactor
+  delegate :params,
+           :edition,
+           :user,
+           :issues,
+           :datetime,
+           to: :context
+
+  def call
+    Edition.transaction do
+      find_and_lock_edition
+      check_for_issues
+      set_scheduled_publishing_datetime
+    end
+  end
+
+private
+
+  def find_and_lock_edition
+    context.edition = Edition.lock.find_current(document: params[:document])
+  end
+
+  def check_for_issues
+    unless edition.draft? || edition.submitted_for_review?
+      raise "Can't save a scheduling date unless edition is a draft or has been submitted for 2i"
+    end
+
+    checker = Requirements::ScheduledDatetimeChecker.new(params[:schedule])
```

Comment:
To save this failing on a potential nil, and other request nasties, I think this could have a method of schedule_params with a `params.require(:schedule).permit()` with the available fields.

Reply:
Done 👍